test/test_reductions.py

```diff
@@ -2180,6 +2180,57 @@ def test_against_np(tensor, bins=100, min=0, max=0):
         expanded = torch.randn(1, 5, 1, 2, device=device).expand(3, 5, 7, 2)
         test_against_np(expanded)
 
+    def test_tensor_compare_ops_empty(self, device):
+        shape = (2, 0, 4)
+        master_input = torch.randn(shape, device=device)
+        test_functions = [
+            ('amax', torch.amax, {}),
+            ('amin', torch.amin, {}),
+            ('argmax', torch.argmax, {'dtype': torch.int64}),
+            ('argmin', torch.argmin, {'dtype': torch.int64}),
+            ('max', lambda *args, **kwargs: torch.max(*args, **kwargs).values, {}),
+            ('min', lambda *args, **kwargs: torch.min(*args, **kwargs).values, {}),
+            ('kthvalue', lambda *args, **kwargs: torch.kthvalue(*args, k=1, **kwargs).values, {}),
+            ('median', lambda *args, **kwargs: torch.median(*args, **kwargs).values, {}),
+        ]
+
+        for name, fn, dtype in test_functions:
+            # Check if reduction happens along the specified dim with and without keepdim.
+            self.assertEqual(torch.empty((2, 0), device=device, **dtype), fn(master_input, dim=2))
+            self.assertEqual(torch.empty((2, 0, 1), device=device, **dtype), fn(master_input, dim=2, keepdim=True))
```

Comment:
It could be a good idea to also test negative `dim` arguments, just to make sure they're being wrapped properly before any of the checks you added

Reply:
Done.